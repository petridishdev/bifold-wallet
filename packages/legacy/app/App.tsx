import {
  AgentProvider,
  AnimatedComponentsProvider,
  AuthProvider,
  ConfigurationProvider,
  ErrorModal,
  NetInfo,
  NetworkProvider,
  RootStack,
  StoreProvider,
  ThemeProvider,
  TourProvider,
  animatedComponents,
  credentialOfferTourSteps,
  credentialsTourSteps,
  defaultConfiguration,
  homeTourSteps,
  initLanguages,
  initStoredLanguage,
  proofRequestTourSteps,
  theme,
  toastConfig,
  translationResources,
} from '@hyperledger/aries-bifold-core'
import * as React from 'react'
import { useEffect, useMemo } from 'react'
import { StatusBar } from 'react-native'
import { isTablet } from 'react-native-device-info'
import Orientation from 'react-native-orientation-locker'
import SplashScreen from 'react-native-splash-screen'
import Toast from 'react-native-toast-message'

initLanguages(translationResources)

const App = () => {
  useMemo(() => {
    initStoredLanguage().then()
  }, [])

  useEffect(() => {
    // Hide the native splash / loading screen so that our
    // RN version can be displayed.
    SplashScreen.hide()
  }, [])

  if (!isTablet()) {
    Orientation.lockToPortrait()
  }

  return (
    <StoreProvider>
      <AgentProvider agent={undefined}>
        <ThemeProvider value={theme}>
          <AnimatedComponentsProvider value={animatedComponents}>
            <ConfigurationProvider value={defaultConfiguration}>
              <AuthProvider>
                <NetworkProvider>
                  <StatusBar
                    hidden={false}
                    barStyle="light-content"
                    backgroundColor={theme.ColorPallet.brand.primary}
                    translucent={false}
                  />
                  <NetInfo />
                  <ErrorModal />
                  <TourProvider
                    homeTourSteps={homeTourSteps}
                    credentialsTourSteps={credentialsTourSteps}
                    credentialOfferTourSteps={credentialOfferTourSteps}
                    proofRequestTourSteps={proofRequestTourSteps}
                    overlayColor={'gray'}
                    overlayOpacity={0.7}
                  >
                    <RootStack />
                  </TourProvider>
                  <Toast topOffset={15} config={toastConfig} />
                </NetworkProvider>
              </AuthProvider>
            </ConfigurationProvider>
          </AnimatedComponentsProvider>
        </ThemeProvider>
      </AgentProvider>
    </StoreProvider>
  )
}

export default App
