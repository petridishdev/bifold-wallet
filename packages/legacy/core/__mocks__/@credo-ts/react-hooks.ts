/* eslint-disable @typescript-eslint/no-explicit-any */
import { LegacyIndyCredentialFormat } from '@credo-ts/anoncreds'
import {
  BasicMessageRecord,
  CredentialExchangeRecord,
  CredentialProtocolOptions,
  ProofExchangeRecord,
  ConnectionRecord,
  DidExchangeRole,
  DidExchangeState,
} from '@credo-ts/core'

const useCredentials = jest.fn().mockReturnValue({ records: [] } as any)
const useProofs = jest.fn().mockReturnValue({ records: [] } as any)
const useCredentialByState = jest.fn().mockReturnValue([] as CredentialExchangeRecord[])
const useProofByState = jest.fn().mockReturnValue([] as ProofExchangeRecord[])
const useBasicMessagesByConnectionId = jest.fn().mockReturnValue([] as BasicMessageRecord[])
const useBasicMessages = jest.fn().mockReturnValue({ records: [] as BasicMessageRecord[] })
const mockCredentialModule = {
  acceptOffer: jest.fn(),
  declineOffer: jest.fn(),
  getFormatData: jest
    .fn()
    .mockReturnValue(
      Promise.resolve({} as CredentialProtocolOptions.GetCredentialFormatDataReturn<[LegacyIndyCredentialFormat]>)
    ),
  findAllByQuery: jest.fn().mockReturnValue(Promise.resolve([])),
}
const mockProofModule = {
  getCredentialsForRequest: jest.fn(),
  acceptRequest: jest.fn(),
  declineRequest: jest.fn(),
  getFormatData: jest.fn(),
  findRequestMessage: jest.fn(),
  requestProof: jest.fn(),
  update: jest.fn(),
  findAllByQuery: jest.fn().mockReturnValue(Promise.resolve([])),
}
const mockBasicMessagesModule = {
  findAllByQuery: jest.fn().mockReturnValue(Promise.resolve([])),
}
const mockConnectionsModule = {
  getAll: jest.fn().mockReturnValue(
    Promise.resolve([
      new ConnectionRecord({
        id: '1',
        did: '9gtPKWtaUKxJir5YG2VPxX',
        theirLabel: 'Faber',
        role: DidExchangeRole.Responder,
        theirDid: '2SBuq9fpLT8qUiQKr2RgBe',
        threadId: '1',
        state: DidExchangeState.Completed,
        createdAt: new Date('2020-01-02T00:00:00.000Z'),
      }),
      new ConnectionRecord({
        id: '2',
        did: '2SBuq9fpLT8qUiQKr2RgBe',
        role: DidExchangeRole.Requester,
        theirLabel: 'Bob',
        theirDid: '9gtPKWtaUKxJir5YG2VPxX',
        threadId: '1',
        state: DidExchangeState.Completed,
        createdAt: new Date('2020-01-01T00:00:00.000Z'),
      }),
    ])
  ),
}

const mockMediationRecipient = {
  initiateMessagePickup: jest.fn(),
}
const mockOobModule = {
  findById: jest.fn().mockReturnValue(Promise.resolve(null)),
  createInvitation: jest.fn(),
  toUrl: jest.fn(),
}
const mockBasicMessageRepository = {
  update: jest.fn(),
}
const mockAgentContext = {
  dependencyManager: {
    resolve: jest.fn().mockReturnValue(mockBasicMessageRepository),
  },
}
const useAgent = () => ({
  agent: {
    credentials: mockCredentialModule,
    proofs: mockProofModule,
    basicMessages: mockBasicMessagesModule,
    connections: mockConnectionsModule,
    mediationRecipient: mockMediationRecipient,
    oob: mockOobModule,
    context: mockAgentContext,
    receiveMessage: jest.fn(),
  },
})
const useCredentialById = jest.fn()
const useProofById = jest.fn()
const useConnectionById = jest.fn()
const useConnections = jest.fn()

export {
  useAgent,
  useBasicMessages,
  useBasicMessagesByConnectionId,
  useConnectionById,
  useConnections,
  useCredentialById,
  useCredentialByState,
  useCredentials,
  useProofById,
  useProofByState,
  useProofs,
}
