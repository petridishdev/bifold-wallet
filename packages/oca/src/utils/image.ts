import { CaptureBaseAttributeType } from '@oca/types'
import { ImageSourcePropType } from 'react-native'

export const toImageSource = (source: unknown): ImageSourcePropType => {
  if (typeof source === 'string') {
    return { uri: source as string }
  }
  return source as ImageSourcePropType
}

export const isDataUrl = (value: string | number | null) => {
  return typeof value === 'string' && value.startsWith('data:image/')
}

export const isBinaryType = (type: string | undefined) => {
  return CaptureBaseAttributeType.Binary === type
}
