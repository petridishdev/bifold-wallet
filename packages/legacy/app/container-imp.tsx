import { BaseLogger } from '@credo-ts/core'
import { Container, TokenMapping } from '@hyperledger/aries-bifold-core'
import { DependencyContainer } from 'tsyringe'
export class AppContainer implements Container {
  private _container: DependencyContainer
  private log?: BaseLogger

  public constructor(bifoldContainer: Container, log?: BaseLogger) {
    this._container = bifoldContainer.container.createChildContainer()
    this.log = log
  }

  public get container(): DependencyContainer {
    return this._container
  }

  public init(): Container {
    // eslint-disable-next-line no-console
    this.log?.info(`Initializing App container`)
    // Here you can register any component to override components in core package
    // Example: Replacing button in core with custom button
    // this.container.registerInstance(TOKENS.COMP_BUTTON, Button)
    return this
  }

  public resolve<K extends keyof TokenMapping>(token: K): TokenMapping[K] {
    return this._container.resolve(token) as TokenMapping[K]
  }
  public resolveAll<K extends keyof TokenMapping, T extends K[]>(
    tokens: [...T]
  ): { [I in keyof T]: TokenMapping[T[I]] } {
    return tokens.map((key) => this.resolve(key)!) as { [I in keyof T]: TokenMapping[T[I]] }
  }
}
