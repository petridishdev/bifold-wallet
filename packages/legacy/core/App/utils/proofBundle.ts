import { BaseLogger } from '@credo-ts/core'
import {
  AnonCredsProofRequestTemplatePayload,
  ProofRequestTemplate,
  useProofRequestTemplates,
} from '@hyperledger/aries-bifold-verifier'
import axios, { AxiosError } from 'axios'

import { useConfiguration } from '../contexts/configuration'

const calculatePreviousYear = (yearOffset: number) => {
  const pastDate = new Date()
  pastDate.setFullYear(pastDate.getFullYear() + yearOffset)
  return parseInt(pastDate.toISOString().split('T')[0].replace(/-/g, ''))
}

export const applyTemplateMarkers = (templates: any): any => {
  if (!templates) return templates
  const markerActions: { [key: string]: (param: string) => string } = {
    now: () => Math.floor(new Date().getTime() / 1000).toString(),
    currentDate: (offset: string) => calculatePreviousYear(parseInt(offset)).toString(),
  }
  let templateString = JSON.stringify(templates)
  // regex to find all markers in the template so we can replace them with computed values
  const markers = [...templateString.matchAll(/"@\{(\w+)(?:\((\S*)\))?\}"/gm)]

  markers.forEach((marker) => {
    const markerValue = markerActions[marker[1] as string](marker[2])
    templateString = templateString.replace(marker[0], markerValue)
  })
  return JSON.parse(templateString)
}

export const applyDevRestrictions = (templates: ProofRequestTemplate[]): ProofRequestTemplate[] => {
  return templates.map((temp) => {
    return {
      ...temp,
      payload: {
        ...temp.payload,
        data: (temp.payload as AnonCredsProofRequestTemplatePayload).data.map((data) => {
          return {
            ...data,
            requestedAttributes: data.requestedAttributes?.map((attr) => {
              return {
                ...attr,
                restrictions: [...(attr.restrictions ?? []), ...(attr.devRestrictions ?? [])],
                devRestrictions: [],
              }
            }),
            requestedPredicates: data.requestedPredicates?.map((pred) => {
              return {
                ...pred,
                restrictions: [...(pred.restrictions ?? []), ...(pred.devRestrictions ?? [])],
                devRestrictions: [],
              }
            }),
          }
        }),
      },
    }
  })
}

export interface ProofBundleResolverType {
  resolve: (acceptDevRestrictions: boolean) => Promise<ProofRequestTemplate[] | undefined>
  resolveById: (templateId: string, acceptDevRestrictions: boolean) => Promise<ProofRequestTemplate | undefined>
}

export const useRemoteProofBundleResolver = (
  indexFileBaseUrl: string | undefined,
  log?: BaseLogger
): ProofBundleResolverType => {
  if (indexFileBaseUrl) {
    return new RemoteProofBundleResolver(indexFileBaseUrl, log)
  } else {
    return new DefaultProofBundleResolver()
  }
}

export class RemoteProofBundleResolver implements ProofBundleResolverType {
  private remoteServer
  private templateData: ProofRequestTemplate[] | undefined
  private log?: BaseLogger

  public constructor(indexFileBaseUrl: string, log?: BaseLogger) {
    this.remoteServer = axios.create({
      baseURL: indexFileBaseUrl,
    })
    this.log = log
  }

  public async resolve(acceptDevRestrictions: boolean): Promise<ProofRequestTemplate[] | undefined> {
    if (this.templateData) {
      let templateData = this.templateData

      if (acceptDevRestrictions) {
        templateData = applyDevRestrictions(templateData)
      }

      return Promise.resolve(templateData)
    }

    return this.remoteServer
      .get('proof-templates.json')
      .then((response) => {
        this.log?.info('Fetched proof templates')

        try {
          let templateData: ProofRequestTemplate[] = response.data
          this.templateData = templateData

          if (acceptDevRestrictions) {
            templateData = applyDevRestrictions(templateData)
          }

          return templateData
        } catch (error: unknown) {
          this.log?.error('Failed to parse proof templates', error as Error)

          return undefined
        }
      })
      .catch((error: AxiosError) => {
        this.log?.error('Failed to fetch proof templates', error)

        return undefined
      })
  }

  public async resolveById(
    templateId: string,
    acceptDevRestrictions: boolean
  ): Promise<ProofRequestTemplate | undefined> {
    if (!this.templateData) {
      return (await this.resolve(acceptDevRestrictions))?.find((template) => template.id === templateId)
    }

    let templateData = this.templateData

    if (acceptDevRestrictions) {
      templateData = applyDevRestrictions(templateData)
    }
    const template = templateData.find((template) => template.id === templateId)

    return template
  }
}

export class DefaultProofBundleResolver implements ProofBundleResolverType {
  private proofRequestTemplates

  public constructor() {
    const { proofRequestTemplates } = useConfiguration()
    this.proofRequestTemplates = proofRequestTemplates ?? useProofRequestTemplates
  }

  public async resolve(acceptDevRestrictions: boolean): Promise<ProofRequestTemplate[]> {
    return Promise.resolve(this.proofRequestTemplates(acceptDevRestrictions))
  }

  public async resolveById(
    templateId: string,
    acceptDevRestrictions: boolean
  ): Promise<ProofRequestTemplate | undefined> {
    return Promise.resolve(
      this.proofRequestTemplates(acceptDevRestrictions).find((template) => template.id === templateId)
    )
  }
}
