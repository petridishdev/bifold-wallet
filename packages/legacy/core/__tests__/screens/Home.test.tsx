import {
  BasicMessageRecord,
  BasicMessageRole,
  CredentialExchangeRecord as CredentialRecord,
  CredentialRole,
  CredentialState,
  ProofExchangeRecord,
  ProofRole,
  ProofState,
} from '@credo-ts/core'
import { useNavigation } from '@react-navigation/native'
import { act, fireEvent, render } from '@testing-library/react-native'

// import { useAgent } from '@credo-ts/react-hooks'

import React from 'react'

// eslint-disable-next-line import/no-named-as-default
import Home from '../../App/screens/Home'
import { testIdWithKey } from '../../App/utils/testable'
import { BasicAppContext } from '../helpers/app'
import { useProofByState, useBasicMessages, useConnectionById, useCredentialByState, useAgent } from '../../__mocks__/@credo-ts/react-hooks'

describe('displays a home screen', () => {
  beforeEach(() => { })

  test('renders correctly', () => {
    const tree = render(
      <BasicAppContext>
        <Home route={{} as any} navigation={useNavigation()} />
      </BasicAppContext>
    )

    expect(tree).toMatchSnapshot()
  })

  /**
   * Scenario: Home Screen without any pending notification
   * Given wallet has successfully loaded
   * When the holder selects the "Home" button in the main navigation bar
   * Then the Home Screen is displayed
   * TODO:(jl) Good enough to be captured by the snapshot?
   */
  test('defaults to no notifications', async () => {
    const { findByText } = render(
      <BasicAppContext>
        <Home route={{} as any} navigation={useNavigation()} />
      </BasicAppContext>
    )
    const notificationLabel = await findByText('Home.NoNewUpdates')

    expect(notificationLabel).toBeTruthy()
  })
})

describe('with a notifications module, when an issuer sends a credential offer', () => {
  const testCredentialRecords: CredentialRecord[] = [
    new CredentialRecord({
      role: CredentialRole.Holder,
      threadId: '1',
      state: CredentialState.OfferReceived,
      protocolVersion: 'v1',
    }),
  ]
  const testProofRecords: ProofExchangeRecord[] = [
    new ProofExchangeRecord({
      role: ProofRole.Prover,
      threadId: '2',
      state: ProofState.RequestReceived,
      protocolVersion: 'v1',
    }),
    new ProofExchangeRecord({
      role: ProofRole.Prover,
      threadId: '3',
      state: ProofState.Done,
      protocolVersion: 'v1',
    }),
  ]
  const testBasicMessages: BasicMessageRecord[] = [
    new BasicMessageRecord({
      threadId: '1',
      connectionId: '1',
      role: BasicMessageRole.Receiver,
      content: 'Hello',
      sentTime: '20200303',
    }),
    new BasicMessageRecord({
      threadId: '2',
      connectionId: '1',
      role: BasicMessageRole.Receiver,
      content: 'Hi',
      sentTime: '20200303',
    }),
  ]

  beforeEach(() => {
    jest.resetAllMocks()
    useBasicMessages.mockReturnValue({ records: testBasicMessages })

    useProofByState.mockReturnValue(testProofRecords)
    useCredentialByState.mockReturnValue(testCredentialRecords)

    useAgent.mockReturnValue({})


    useConnectionById.mockReturnValue({ theirLabel: 'ACME' })
  })

  /**
   * Scenario: Holder receives a credential offer
   * Given the holder has a connection with an issuer
   * When the issuer sends a credential offer
   * Then the credential offer will arrive in the form of a notification in the home screen
   */
  test('notifications are displayed', async () => {
    const { findAllByTestId } = render(
      <BasicAppContext>
        <Home route={{} as any} navigation={useNavigation()} />
      </BasicAppContext>
    )

    const flatListInstance = await findAllByTestId(testIdWithKey('NotificationListItem'))

    expect(flatListInstance).toHaveLength(4)
  })

  /**
   * Scenario: Holder selects a credential offer
   * Given the holder has received a credential offer from a connected issuer
   * When the holder selects the credential offer
   * When the holder is taken to the credential offer screen/flow
   */
  test('touch notification triggers navigation correctly I', async () => {
    const navigation = useNavigation()
    const view = render(
      <BasicAppContext>
        <Home route={{} as any} navigation={useNavigation()} />
      </BasicAppContext>
    )

    const button = await view.findByTestId(testIdWithKey('ViewOffer'))

    expect(button).toBeDefined()
    //view.debug()
    await act(() => {
      fireEvent(button, 'press')
    })

    expect(navigation.navigate).toHaveBeenCalledTimes(1)
    expect(navigation.navigate).toHaveBeenCalledWith('Notifications Stack', {
      screen: 'Credential Offer',
      params: { credentialId: testCredentialRecords[0].id },
    })
  })

  /**
   * Scenario: Holder selects a proof request
   * Given the holder has received a proof request from a connected verifier
   * When the holder selects the proof request
   * When the holder is taken to the proof request screen/flow
   */
  test('touch notification triggers navigation correctly II', async () => {
    const { findByTestId } = render(
      <BasicAppContext>
        <Home route={{} as any} navigation={useNavigation()} />
      </BasicAppContext>
    )

    const button = await findByTestId(testIdWithKey('ViewProofRecord'))
    const navigation = useNavigation()

    expect(button).toBeDefined()

    await act(() => {
      fireEvent(button, 'press')
    })

    expect(navigation.navigate).toHaveBeenCalledTimes(1)

    expect(navigation.navigate).toHaveBeenCalledWith('Notifications Stack', {
      screen: 'Proof Request',
      params: { proofId: testProofRecords[0].id },
    })
  })

  /**
   * Scenario: Holder selects a proof request
   * Given the holder has received a proof request from a connected verifier
   * When the holder selects the proof request
   * When the holder is taken to the proof request screen/flow
   */
  test('touch notification triggers navigation correctly III', async () => {
    const { findByTestId } = render(
      <BasicAppContext>
        <Home route={{} as any} navigation={useNavigation()} />
      </BasicAppContext>
    )

    const button = await findByTestId(testIdWithKey('ViewProofRecordReceived'))
    const navigation = useNavigation()

    expect(button).toBeDefined()

    act(() => {
      fireEvent(button, 'press')
    })

    expect(navigation.navigate).toHaveBeenCalledTimes(1)
    expect(navigation.navigate).toHaveBeenCalledWith('Contacts Stack', {
      screen: 'Proof Details',
      params: { recordId: testProofRecords[1].id, isHistory: true },
    })
  })

  /**
   * Scenario: Holder selects a basic message notification
   * Given the holder has received a message from a contact
   * When the holder taps the View message button
   * The holder is taken to the chat screen for that contact
   */
  test('touch notification triggers navigation correctly IV', async () => {
    const { findAllByTestId } = render(
      <BasicAppContext>
        <Home route={{} as any} navigation={useNavigation()} />
      </BasicAppContext>
    )

    await act(async () => {
      const button = (await findAllByTestId(testIdWithKey('ViewBasicMessage')))[0]
      expect(button).toBeDefined()
      fireEvent(button, 'press')
    })

    //xxx
    const navigation = useNavigation()
    expect(navigation.getParent()?.navigate).toHaveBeenCalledTimes(1)
    expect(navigation.getParent()?.navigate).toHaveBeenCalledWith('Contacts Stack', {
      screen: 'Chat',
      params: { connectionId: '1' },
    })
  })
})
