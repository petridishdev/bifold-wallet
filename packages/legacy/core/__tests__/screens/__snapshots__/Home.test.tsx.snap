// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays a home screen renders correctly 1`] = `
<RCTScrollView
  ListEmptyComponent={[Function]}
  ListFooterComponent={[Function]}
  ListHeaderComponent={[Function]}
  data={Array []}
  decelerationRate="fast"
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEnabled={false}
  scrollEventThrottle={50}
  showsVerticalScrollIndicator={false}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "marginBottom": 35,
    }
  }
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      collapsable={false}
      onLayout={[Function]}
    >
      <View />
    </View>
    <View
      style={
        Object {
          "backgroundColor": "#313132",
          "paddingHorizontal": 20,
          "paddingVertical": 20,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#313132",
            "borderColor": "#0099FF",
            "borderRadius": 5,
            "borderWidth": 1,
            "padding": 10,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
            }
          }
        >
          <View
            style={
              Object {
                "alignSelf": "flex-start",
                "marginLeft": 0,
                "marginRight": 10,
              }
            }
          >
            <Text
              allowFontScaling={false}
              selectable={false}
              style={
                Array [
                  Object {
                    "color": "#0099FF",
                    "fontSize": 30,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              î¢Ž
            </Text>
          </View>
          <Text
            style={
              Object {
                "alignSelf": "center",
                "color": "#FFFFFF",
                "flex": 1,
                "flexWrap": "wrap",
                "fontSize": 18,
                "fontWeight": "normal",
              }
            }
            testID="com.ariesbifold:id/NoNewUpdates"
          >
            Home.NoNewUpdates
          </Text>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
    >
      <View>
        <View
          style={
            Object {
              "paddingBottom": 75,
              "paddingHorizontal": 25,
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "marginHorizontal": 25,
                  "marginTop": 35,
                },
              ]
            }
          >
            <Text
              adjustsFontSizeToFit={true}
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 38,
                    "fontWeight": "bold",
                  },
                  Object {
                    "marginBottom": 20,
                    "marginTop": 25,
                  },
                ]
              }
            >
              Home.Welcome
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                  "marginHorizontal": 25,
                  "marginTop": 35,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 18,
                    "fontWeight": "normal",
                  },
                  Object {
                    "marginTop": 25,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Home.NoCredentials
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
